src/transformers/generation/logits_process.py

```diff
@@ -1399,6 +1401,8 @@ class ClassifierFreeGuidanceLogitsProcessor(LogitsProcessor):
     correspond to the unconditional logits (predicted from an empty or 'null' prompt). The processor computes a
     weighted average across the conditional and unconditional logits, parameterised by the `guidance_scale`.
 
+    See [the paper](https://arxiv.org/abs/2306.17806) for more information.
```

Comment:
This is also not the paper used here :)

The correct paper is https://arxiv.org/abs/2306.05284 (similar techniques, but not exactly the same, the one you linked can be seen as an upgrade to this technique)